<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>750</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget_editor">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>491</width>
      <height>461</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget_output">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>460</y>
      <width>491</width>
      <height>221</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>491</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_problems">
     <attribute name="title">
      <string>Problems</string>
     </attribute>
     <widget class="QListWidget" name="listWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>483</width>
        <height>186</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::HLine</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_console">
     <attribute name="title">
      <string>Console</string>
     </attribute>
     <widget class="QPlainTextEdit" name="plainTextEdit">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>483</width>
        <height>186</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_screen">
     <attribute name="title">
      <string>Screen</string>
     </attribute>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>0</y>
        <width>256</width>
        <height>185</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Panel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>0</y>
      <width>511</width>
      <height>681</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetNoConstraint</enum>
     </property>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget_pipeline">
       <property name="styleSheet">
        <string notr="true">QTabBar::tab { height: 110px; }</string>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::West</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Triangular</enum>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <property name="tabBarAutoHide">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="tab_scalar">
        <attribute name="title">
         <string>Scalar</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tab_vector">
        <attribute name="title">
         <string>Vector</string>
        </attribute>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget_memory">
       <property name="currentIndex">
        <number>3</number>
       </property>
       <widget class="QWidget" name="tab_register">
        <attribute name="title">
         <string>Register</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_6">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Binary</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Decimal</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Hex</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ASC II</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_flag">
        <attribute name="title">
         <string>Flag</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_8">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Binary</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Decimal</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Hex</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ASC II</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_vector">
        <attribute name="title">
         <string>Vector</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_7">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Binary</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Decimal</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Hex</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ASC II</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_memory">
        <attribute name="title">
         <string>Memory</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_9">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>191</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Binary</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Decimal</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Hex</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ASC II</string>
          </property>
         </column>
        </widget>
        <widget class="QLineEdit" name="lineEdit">
         <property name="geometry">
          <rect>
           <x>400</x>
           <y>160</y>
           <width>101</width>
           <height>29</height>
          </rect>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget_cache">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_cache_l1_i">
        <attribute name="title">
         <string>Cache L1(I)</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Tag</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Set</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>LRU</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word0</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word3</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_cache_l1_d">
        <attribute name="title">
         <string>Cache L1(D)</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Tag</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Set</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>LRU</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word0</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word3</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_cache_l2">
        <attribute name="title">
         <string>Cache L2</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_4">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Tag</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Set</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>LRU</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word0</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word3</string>
          </property>
         </column>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_cache_l3">
        <attribute name="title">
         <string>Cache L3</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_5">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>501</width>
           <height>192</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <column>
          <property name="text">
           <string>Tag</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Set</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>LRU</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word0</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Word3</string>
          </property>
         </column>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
   </widget>
   <widget class="QMenu" name="menuEdt">
    <property name="title">
     <string>E&amp;dt</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionDelete"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>V&amp;iew</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuWindow_2">
    <property name="title">
     <string>Wi&amp;ndow</string>
    </property>
    <addaction name="actionCache_View"/>
    <addaction name="actionPipeline"/>
    <addaction name="actionCode_Examiner"/>
    <addaction name="actionMemory"/>
    <addaction name="actionRegister"/>
    <addaction name="actionConsole"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionPISA_Help"/>
    <addaction name="actionAbout_PISA"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdt"/>
   <addaction name="menuWindow"/>
   <addaction name="menuWindow_2"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionnew"/>
   <addaction name="actionopen"/>
   <addaction name="actionSave_2"/>
   <addaction name="separator"/>
   <addaction name="actionRun"/>
   <addaction name="actionpause"/>
   <addaction name="actionstop"/>
   <addaction name="separator"/>
   <addaction name="actionbuild"/>
   <addaction name="actionbuild_all"/>
   <addaction name="separator"/>
   <addaction name="actionrewind"/>
   <addaction name="actionforward"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Sa&amp;ve As</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="actionCache_View">
   <property name="text">
    <string>&amp;Cache</string>
   </property>
  </action>
  <action name="actionPipeline">
   <property name="text">
    <string>&amp;Pipeline</string>
   </property>
  </action>
  <action name="actionCode_Examiner">
   <property name="text">
    <string>Code &amp;Examiner</string>
   </property>
  </action>
  <action name="actionMemory">
   <property name="text">
    <string>&amp;Memory</string>
   </property>
  </action>
  <action name="actionRegister">
   <property name="text">
    <string>&amp;Register</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/play.png</normaloff>:/Menu_icon/play.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Run Program</string>
   </property>
  </action>
  <action name="actionConsole">
   <property name="text">
    <string>C&amp;onsole</string>
   </property>
  </action>
  <action name="actionAbout_PISA">
   <property name="text">
    <string>&amp;About PISA</string>
   </property>
  </action>
  <action name="actionPISA_Help">
   <property name="text">
    <string>&amp;PISA Help</string>
   </property>
  </action>
  <action name="actionnew">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/new.png</normaloff>:/Menu_icon/new.png</iconset>
   </property>
   <property name="text">
    <string>new</string>
   </property>
   <property name="toolTip">
    <string>Create New File</string>
   </property>
  </action>
  <action name="actionopen">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/open.png</normaloff>:/Menu_icon/open.png</iconset>
   </property>
   <property name="text">
    <string>open</string>
   </property>
   <property name="toolTip">
    <string>Open File</string>
   </property>
  </action>
  <action name="actionSave_2">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/save.png</normaloff>:/Menu_icon/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save File</string>
   </property>
  </action>
  <action name="actionstop">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/stop.png</normaloff>:/Menu_icon/stop.png</iconset>
   </property>
   <property name="text">
    <string>stop</string>
   </property>
   <property name="toolTip">
    <string>Stop Program</string>
   </property>
  </action>
  <action name="actionpause">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/pause.png</normaloff>:/Menu_icon/pause.png</iconset>
   </property>
   <property name="text">
    <string>pause</string>
   </property>
   <property name="toolTip">
    <string>Pause Program</string>
   </property>
  </action>
  <action name="actionforward">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/forward.png</normaloff>:/Menu_icon/forward.png</iconset>
   </property>
   <property name="text">
    <string>forward</string>
   </property>
   <property name="toolTip">
    <string>Forward Program</string>
   </property>
  </action>
  <action name="actionrewind">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/backward.png</normaloff>:/Menu_icon/backward.png</iconset>
   </property>
   <property name="text">
    <string>rewind</string>
   </property>
   <property name="toolTip">
    <string>Rewind Program</string>
   </property>
  </action>
  <action name="actionbuild">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/build.png</normaloff>:/Menu_icon/build.png</iconset>
   </property>
   <property name="text">
    <string>build</string>
   </property>
   <property name="toolTip">
    <string>Build Current File</string>
   </property>
  </action>
  <action name="actionbuild_all">
   <property name="icon">
    <iconset resource="rsc/pisa_resource.qrc">
     <normaloff>:/Menu_icon/build_all.png</normaloff>:/Menu_icon/build_all.png</iconset>
   </property>
   <property name="text">
    <string>build all</string>
   </property>
   <property name="toolTip">
    <string>Build All Files</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>&amp;Cut</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>C&amp;opy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>&amp;Delete</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>tabWidget_output</tabstop>
  <tabstop>tabWidget_editor</tabstop>
  <tabstop>tabWidget_pipeline</tabstop>
  <tabstop>tabWidget_cache</tabstop>
  <tabstop>tabWidget_memory</tabstop>
 </tabstops>
 <resources>
  <include location="rsc/pisa_resource.qrc"/>
 </resources>
 <connections/>
</ui>
